share/config.py

```diff
@@ -264,3 +272,107 @@ def parse_config(config_yaml: str) -> Config:
+
+        if secret not in needed_secret_arns:
+            needed_secret_arns[secret] = ""
+
+        region = splitted_secret[3]
+        secrets_manager_name = splitted_secret[6]
+
+        if region == "":
+            raise ValueError("Must be provided region in arn: {}".format(secret))
+
+        if secrets_manager_name == "":
+            raise ValueError("Must be provided secrets manager name in arn: {}".format(secret))
+
+        if region not in needed_secret_keys:
+            needed_secret_keys[region] = {}
+
+        if secrets_manager_name not in needed_secret_keys[region]:
+            needed_secret_keys[region][secrets_manager_name] = ""
+
+    for region in needed_secret_keys:
+        for secrets_manager_name in needed_secret_keys[region]:
+            str_secrets = get_secret_values(secrets_manager_name, region)
+            parsed_secrets = parse_secrets_str(str_secrets)
+
+            needed_secret_keys[region][secrets_manager_name] = parsed_secrets
+
+    for secret in needed_secret_arns:
+        splitted_secret_string = secret.split(":")
+
+        region = splitted_secret_string[3]
+        secrets_manager_name = splitted_secret_string[6]
+
+        if len(splitted_secret_string) == 8:
+            wanted_key = splitted_secret_string[-1]
+
+            for got_key, got_value in needed_secret_keys[region][secrets_manager_name].items():
+
+                if got_key == wanted_key:
+                    needed_secret_arns[secret] = got_value
+
+        else:
+            needed_secret_arns[secret] = needed_secret_keys[region][secrets_manager_name]
+
+    for secret in needed_secret_arns:
+        if needed_secret_arns[secret] == "":
+            raise ValueError("Value for key: {} must not be empty".format(secret))
+
+        config_yaml = config_yaml.replace(secret, needed_secret_arns[secret])
+
+    return config_yaml
+
+
+def get_secret_values(secret_name: str, region_name: str) -> str:
+    session = boto3.session.Session()
+    client = session.client(service_name="secretsmanager", region_name=region_name)
+
+    try:
+        get_secret_value_response = client.get_secret_value(SecretId=secret_name)
+    except ClientError as e:
+        if e.response["Error"]["Code"] == "DecryptionFailureException":
+            raise e
+        elif e.response["Error"]["Code"] == "InternalServiceErrorException":
+            raise e
+        elif e.response["Error"]["Code"] == "InvalidParameterException":
+            raise e
+        elif e.response["Error"]["Code"] == "InvalidRequestException":
+            raise e
+        elif e.response["Error"]["Code"] == "ResourceNotFoundException":
+            raise KeyError("Cannot find secret: {}".format(secret_name)) from None
+    else:
+        if "SecretString" in get_secret_value_response:
+            secrets = get_secret_value_response["SecretString"]
+
+        else:
+            secrets = get_secret_value_response["SecretBinary"].decode("utf-8")
```

Comment:
this should be base64 decoded, shouldn't it?

Reply:
it is correct that in the documentation it says it will be a base64 format, but when I tested it, it comes as a b"string". I will do more tests for this. Maybe I'm not creating the binary secret correctly.

Reply:
maybe a base64 encoded as string type?
what's the content of the secret?

Reply:
accessing this: `get_secret_value_response["SecretBinary"]`, will output `b"this_is_my_secret"`. It's a bytes string. I don't need to decode from base64

Reply:
how was defined the secret?
can you try forcing a binary one with aws cli, thanks
```
aws secretsmanager put-secret-value --secret-id xxx --secret-binary fileb://asecret.file
```

Reply:
I defined the secret with create_secret api from boto3, but I also tried your way and it's still the same.. it returns the byte string

Reply:
https://github.com/boto/boto3/issues/2735

indeed boto3 does the decoding